File: lib/core/config/app_config.dart
class AppConfig {
  AppConfig._();

  static const env = String.fromEnvironment('APP_ENV', defaultValue: 'dev');

  static bool get isProd => env == 'prod';
  static bool get isDev => !isProd;

  static String get driftDbName =>
      isProd ? 'salik_crm_local_prod' : 'salik_crm_local_dev';

  static String get envLabel => isProd ? 'PROD' : 'DEV';

  static String get envDisplayName => isProd ? 'Production App' : 'Dev App';

  static String get firebaseProjectId =>
      isProd ? 'salikeencms-prod' : 'salikeencms';

  static String get staffEmailDomain =>
      isProd ? '@cms.com' : '@dev.cms.com';

  /// Shared AES-256 field-crypto key (base64, 32 bytes). Same value for .NET.
  /// Override via `--dart-define=FIELD_CRYPTO_KEY_BASE64=...`
  static const fieldCryptoKeyBase64 = String.fromEnvironment(
    'FIELD_CRYPTO_KEY_BASE64',
    defaultValue: 'oseHVKBqsJE6twGWGPl4/51tjxNknNktqe1wU6vT1Ws=',
  );
}

File: app/src/main/java/com/example/salik_management_system/core/config/AppConfig.kt
package com.example.salik_management_system.core.config

import com.example.salik_management_system.BuildConfig

object AppConfig {
    val env: String
        get() = BuildConfig.APP_ENV

    val isProd: Boolean
        get() = env.equals("prod", ignoreCase = true)

    val isDev: Boolean
        get() = !isProd

    /** Room database file name (parity with Flutter driftDbName). */
    val roomDbName: String
        get() = if (isProd) "salik_crm_local_prod" else "salik_crm_local_dev"

    /** Internal / log label. */
    val envLabel: String
        get() = if (isProd) "prod" else "dev"

    /** User-facing chip — only Dev shows a badge. */
    val showDevBadge: Boolean
        get() = isDev

    val envDisplayName: String
        get() = if (isProd) "Salikeen CMS" else "Dev App"

    val firebaseProjectId: String
        get() = if (isProd) "salikeencms-prod" else "salikeencms"

    val staffEmailDomain: String
        get() = if (isProd) "@cms.com" else "@dev.cms.com"

    /** Shared AES-256 field-crypto key (base64, 32 bytes). */
    val fieldCryptoKeyBase64: String
        get() = BuildConfig.FIELD_CRYPTO_KEY_BASE64
}
